src/ray/raylet/worker_pool.cc

```diff
@@ -416,7 +428,11 @@ Process WorkerPool::StartWorkerProcess(
                                worker_type);
   state.starting_worker_processes.emplace(
       worker_startup_token_counter_,
-      StartingWorkerProcessInfo{workers_to_start, workers_to_start, worker_type, proc});
+      StartingWorkerProcessInfo{workers_to_start, workers_to_start, worker_type, proc,
+                                runtime_env_info});
+  runtime_env_manager_.AddURIReference(
```

Comment:
Can we put `state.starting_worker_processes.emplace(` logic and `runtime_env_manager_.AddURIReference(` in the same private method? Seems like they should always be coupled right? Same for `.erase` and `RemoveURIReference `

Reply:
A little weird because this private method only used by `StartWorkerProcess`. But the erase logic is used in two places now. Make sense overall.

Reply:
I think this will help us from committing mistakes, such as somebody add code such as `state.starting_worker_processes.emplace` without adding uri reference. Not sure if the private method is the best choice, but I think this 2 functions should be coupled always. 

Maybe we can just make `OnProcessStarted`, `OnWorkerStarted`, `OnProcessRemoved`?

Reply:
I have added two functions https://github.com/ray-project/ray/pull/20789/commits/aba42f88018dc314fdde85c303ea1b52c432cbef